TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs

```diff
@@ -158,7 +158,31 @@ private IEnumerable<MailAddress> GetAddresses(string key)
     /// <param name="addresses">The reply-to addresses.</param>
     public void AddReplyTo(IEnumerable<MailAddress> addresses) => ReplyTo = ReplyTo.Concat(addresses);
 
+    /// <summary>
+    /// Gets the headers that are not mail addresses and therefore have no getter/setter.
+    /// </summary>
     public IReadOnlyDictionary<string, object> NonAddressHeaders => this
         .ExceptBy(AddressKeys, x => x.Key)
         .ToDictionary(x => x.Key, x => x.Value);
+
+    /// <summary>
+    /// Appends <paramref name="other"/> to this collection without overwriting existing values.
+    /// </summary>
+    /// <param name="other">The collection to append.</param>
+    /// <returns>The current instance.</returns>
+    public MailHeaderCollection Append(MailHeaderCollection other)
```

Comment:
I wonder if `bool overwrite = false` would be useful, in the spirit of letting others use this as well.

Reply:
I added a separate method instead